src/server/json_family.cc

```diff
@@ -246,16 +253,144 @@ OpResult<vector<OptBool>> OpToggle(const OpArgs& op_args, string_view key, strin
     return OpStatus::SYNTAX_ERR;
   }
 
-  auto [it_output, added] = db_slice.AddOrFind(op_args.db_ind, key);
-  it_output->second.SetString(j.as_string());
-  db_slice.PostUpdate(op_args.db_ind, it_output);
-  RecordJournal(op_args, it_output->first, it_output->second);
+  SetString(op_args, key, j.as_string());
+  return vec;
+}
+
+OpResult<vector<OptLongLong>> OpNumIncrBy(const OpArgs& op_args, string_view key, string_view path, long long num) {
+  OpResult<json> result = GetJson(op_args, key);
+  if (!result) {
+    return result.status();
+  }
+
+  vector<OptLongLong> vec;
+  auto cb = [&](const string& path, json& val) {
+    if (val.is_number()) {
+      auto current_val = val.as<long long>() + num;
+      val = current_val;
+      vec.emplace_back(current_val);
+    } else {
+      vec.emplace_back(nullopt);
+    }
+  };
+
+  json j = result.value();
+  error_code ec = JsonReplace(j, path, cb);
+  if (ec) {
+    VLOG(1) << "Failed to evaulate expression on json with error: " << ec.message();
+    return OpStatus::SYNTAX_ERR;
+  }
 
+  SetString(op_args, key, j.as_string());
+  return vec;
+}
+
+OpResult<vector<OptLongLong>> OpNumMultBy(const OpArgs& op_args, string_view key, string_view path, long long num) {
+  OpResult<json> result = GetJson(op_args, key);
+  if (!result) {
+    return result.status();
+  }
+
+  vector<OptLongLong> vec;
+  auto cb = [&](const string& path, json& val) {
+    if (val.is_number()) {
+      auto current_val = val.as<long long>() * num;
+      val = current_val;
+      vec.emplace_back(current_val);
+    } else {
+      vec.emplace_back(nullopt);
+    }
+  };
+
+  json j = result.value();
+  error_code ec = JsonReplace(j, path, cb);
+  if (ec) {
+    VLOG(1) << "Failed to evaulate expression on json with error: " << ec.message();
+    return OpStatus::SYNTAX_ERR;
+  }
+
+  SetString(op_args, key, j.as_string());
   return vec;
 }
 
 } // namespace
 
+void JsonFamily::NumIncrBy(CmdArgList args, ConnectionContext* cntx) {
+  string_view key = ArgS(args, 1);
+  string_view path = ArgS(args, 2);
+  string_view num = ArgS(args, 3);
+
+  long long llnum;
+  if (!string2ll(num.data(), num.size(), &llnum)) {
+    (*cntx)->SendError(kWrongTypeErr);
+    return;
+  }
+
+  auto cb = [&](Transaction* t, EngineShard* shard) {
+    return OpNumIncrBy(t->GetOpArgs(shard), key, path, llnum);
+  };
+
+  DVLOG(1) << "Before Get::ScheduleSingleHopT " << key;
+  Transaction* trans = cntx->transaction;
+  OpResult<vector<OptLongLong>> result = trans->ScheduleSingleHopT(move(cb));
+
+  if (result) {
+    DVLOG(1) << "JSON.NUMINCRBY " << trans->DebugId() << ": " << key;
+    if (result->empty()) {
+      (*cntx)->SendNullArray();
+    } else {
+      (*cntx)->StartArray(result->size());
+      for (auto& it: *result) {
+        if (it.has_value()) {
+          (*cntx)->SendLong(*it);
```

Comment:
@romange, Should I implement a function that retrieves a `long long` value, right?

Reply:
you mean like `SendLong`? Then no, because on 64bit architectures `long long` and `long` is the same and we are only 64bit.